src/harness/fakesHosts.ts

```diff
@@ -371,7 +371,7 @@ export class CompilerHost implements ts.CompilerHost {
         // and so any options bag will be keyed as "[object Object]", and we'll incorrectly share
         // SourceFiles parsed with different options. But fixing this doesn't expose any bugs and
         // doubles the memory usage of a test run, so I'm leaving it for now.
-        const cacheKey = this.vfs.shadowRoot && `SourceFile[languageVersionOrOptions=${languageVersionOrOptions},setParentNodes=${this._setParentNodes}]`;
+        const cacheKey = this.vfs.shadowRoot && `SourceFile[languageVersionOrOptions=${languageVersionOrOptions},setParentNodes=${this._setParentNodes},accessPath=${canonicalFileName}]`;

```

Comment:
The problem is that if we have a symlink, then vfs will resolve to the same meta collection even though we are accessing the file through different paths. But the SourceFile also has the file path in it which TSC expects to be the path the file was loaded at.

This was not a problem when all that was cached were lib files. Now since we run the tests multiple times it does help to cache the SourceFile for other source files in the tests and some of them are symlinks.

Reply:
> This was not a problem when all that was cached were lib files. Now since we run the tests multiple times it does help to cache the SourceFile for other source files in the tests and some of them are symlinks.

Yeah, I think the main problem is that reusing files here that are not lib files means that other properties (like the implicit module format) may differ. But, the tests pass.

But, like you said, if the canonical path being different is rare, then things didn't get more buggy.